### PR TITLE
3753: Media Attachable

### PR DESCRIPTION
**Pull request by meltheadorable**
* Renames and modifies `mediable` spec example group to create `media_attachable`
* Adds image for spec assets
*  Changes references to `mediable` to `media_attachable`
* Updates factory for media model
* Adds `carrierwave`, `mini_magick`, and `awesome_print` to Gemfile
* Adds media model
* Makes `media` uncountable to avoid inflection weirdness in rails
* Adds media item uploader
* Creates `MediaAttachable` concern
* Adds upload directory to `.gitignore`

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,6 +55,9 @@ group :development, :test do
   # Report number of queries in server log
   gem 'sql_queries_count'
 
+  # Better console printing
+  gem 'awesome_print'
```
Comment (linesarefuzzy) [inline Gemfile]: :+1:

File: app/uploaders/media_item_uploader.rb [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  #TODO: Identify appropriate dimensions for these versions
+  version :small
+  version :medium
+  version :large
+
+  # the directory where uploaded files will be stored.
+  def store_dir
+    owner_type = model.media_attachable_type.underscore
+    owner_id = model.media_attachable_id
+    path = File.join('uploads', Rails.env, owner_type, "#{owner_id}", "#{model.id}")
```
Comment (linesarefuzzy) [inline app/uploaders/media_item_uploader.rb]: Why the double quotes?

File: app/uploaders/media_item_uploader.rb [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      model.kind = 'file'
+    end
+  end
+
+  def set_height_and_width_on_model
+    if file && model
+      model.item_width, model.item_height = ::MiniMagick::Image.open(file.file)[:dimensions]
+    end
+  end
+
```
Comment (linesarefuzzy) [inline app/uploaders/media_item_uploader.rb]: What's the purpose of setting all these things on the model?

**Comment by linesarefuzzy** [comment]
Looks great. Merge when ready.

File: app/uploaders/media_item_uploader.rb (hunk 2 above)
Comment (meltheadorable) [inline app/uploaders/media_item_uploader.rb]: It was either that or an explicit cast to string, I decided the interpolation was cleaner.

File: app/uploaders/media_item_uploader.rb (hunk 3 above)
Comment (meltheadorable) [inline app/uploaders/media_item_uploader.rb]: By default carrierwave only persists the path to the image, when I needed to opt for it over Paperclip I needed to set and persist them manually. I don't know whether they are or will need to be used, but they were present in the schema doc and most are also present by default in Paperclip do I created them here too.

**Pull request merged**